### ISSUE TITLE
[Bug]: o_versionCount not correctly updated in objects table when versions are disabled. Admin UI then shows wrong data

### ISSUE THREAD

**Issue by chm0815** [pimcore/pimcore#16039]
It seems, there is a serious problem caused by a **race condition**. 
I found one way to reproduce the problem (**objects.o_versionCount < versions.versionCount**) for non-draft objects.

Imagine you have a command that saves objects and has **disabled version creation**.
Eg.
```php
Version::disable();
$obj = DataObject::getById(1234);
// sleep(100) or set a breakpoint. In the meantime, save the same object with enabled versions (eg via the Admin UI)
$obj->save();

```

The problem now is that the object saved with the code above has now a smaller o_versionCount than the newest version in the versions table, because the o_versionCount is only updated when the version creation is enabled [here].https://github.com/pimcore/pimcore/blob/b147e218e55da7cb2bb3023a6b499cda7021f6f9/models/Element/AbstractElement.php#L296.

The object edit UI will potentially display different data than what is stored in the object_* tables what is very misleading for the user, because of [getLatestVersion](https://github.com/pimcore/pimcore/blob/0fbdbff0e54fe65af2225f3f4e30062c11e9b272/models/Element/Traits/VersionDaoTrait.php#L32).  

There was already a similar issue here: https://github.com/pimcore/pimcore/issues/3523

_Originally posted by @chm0815 in https://github.com/pimcore/pimcore/discussions/16022#discussioncomment-7162784_

**Comment by chm0815** [comment]
See Pull-Request: https://github.com/pimcore/pimcore/pull/16120

**Comment by dvesh3** [comment]
Fixed by #16120
